tests/ProxyManagerTest/Factory/AbstractBaseFactoryTest.php

```diff
@@ -169,4 +170,41 @@ public function testGeneratesClass() : void
             $generateProxy->invoke($this->factory, stdClass::class, ['some' => 'proxy', 'options' => 'here'])
         );
     }
+
+    /**
+     * @group 469
+     */
+    public function testGeneratesClassBasedOnGivenProxyOptions() : void
+    {
+        $generateProxy = new ReflectionMethod($this->factory, 'generateProxy');
+
+        $generateProxy->setAccessible(true);
+        $generatedClass = UniqueIdentifierGenerator::getIdentifier('fooBar');
+        $proxyOptions   = [uniqid('key', true) => uniqid('value', true)];
+
+        $this
+            ->classNameInflector
+            ->expects(self::any())
```

Comment:
Why not be more specific here?

Reply:
Not part of the scope of this particular test (tested in other tests in the same class IIRC)